### PR TITLE
Add getters to FFmpegOutputBuilder and deprecate property access

### PR DESCRIPTION
**Pull request by Euklios**
The goal is to allow changes to classes' internal structure and storage layout without affecting their public interface. Getters can be added to return the same information as before, while the internal might keep its information in a different way. I plan to use this in #318 to add details about each option. The getters allow me to create a seamless transition between old and new. Additionally, this makes the API more consistent within the library and other Java code.

### CODE REVIEW

**Comment by bramp** [comment]
and sorry can you briefly describe why you are making this change. What benefit does it have?

File: src/main/java/net/bramp/ffmpeg/builder/AbstractFFmpegStreamBuilder.java [hunk 1]
```diff
@@ -686,4 +749,120 @@ protected void addVideoFlags(FFmpegBuilder parent, ImmutableList.Builder<String>
...
+  public Long getStartOffset() {
+    return startOffset;
+  }
+
+  public Long getDuration() {
+    return duration;
+  }
+
+  public List<String> getMetaTags() {
+    return meta_tags;
```
[the remark this answers is not in the document]
Comment (Euklios) [reply src/main/java/net/bramp/ffmpeg/builder/AbstractFFmpegStreamBuilder.java]: Yes, fixed

**Comment by Euklios** [comment]
> and sorry can you briefly describe why you are making this change. What benefit does it have?

The goal is to allow changes to classes' internal structure and storage layout without affecting their public interface. Getters can be added to return the same information as before, while the internal might keep its information in a different way. I plan to use this in #318 to add details about each option. The getters allow me to create a seamless transition between old and new. Additionally, this makes the API more consistent within the library and other Java code.

**Review by bramp — APPROVED** [review]
Looks good, my only concern is if this is worth it for the simple "Immutable" objects such as `Format`?, e.g

```java
@Immutable
public class Format {
  final String name;
  final String longName;
...
```

They represent a data structure, that isn't going to change (or at least I don't see how it would right now).

But I won't block you from making changes... Either way this seems a API change, so we should bump the version number appropriately.

**Pull request merged**